src/librustdoc/clean/mod.rs

```diff
@@ -1095,86 +1091,64 @@ fn clean_function<'tcx>(
     cx: &mut DocContext<'tcx>,
     sig: &hir::FnSig<'tcx>,
     generics: &hir::Generics<'tcx>,
-    args: FunctionArgs<'tcx>,
+    params: ParamsSrc<'tcx>,
 ) -> Box<Function> {
     let (generics, decl) = enter_impl_trait(cx, |cx| {
-        // NOTE: generics must be cleaned before args
+        // NOTE: Generics must be cleaned before params.
```

Comment:
Maybe we could write a function which cleans generics and params and return them. Like that we're sure that the order is enforced (but please keep the note comment). Since it's done in more than one location, I think it would make sense. What do you think?

Reply:
Well, I'm planning on rewriting bounds cleaning for 1.5 years (WIP PR https://github.com/fmease/rust/pull/23 is stuck in development hell) to fix https://github.com/rust-lang/rust/issues/113015 and https://github.com/rust-lang/rust/issues/126432 which will completely change everything and likely also fix this comment.

I'd rather not change things unrelated to the 3 regressions I'm fixing here. The refactoring commit is merely fixing these comments.